packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -176,12 +175,19 @@ const withDefaultPublicRoutes = (publicRoutes: RouteMatcherParam | undefined) =>
   if (typeof publicRoutes === 'function') {
     return publicRoutes;
   }
+
   const routes = [publicRoutes || ''].flat().filter(Boolean);
-  if (SIGN_IN_URL) {
-    routes.push(matchRoutesStartingWith(SIGN_IN_URL));
+  // TODO: refactor it to use common config file eg SIGN_IN_URL from ./clerkClient
```

Comment:
❓ Don't we do the same thing in clerkClient.ts?

Reply:
what do you mean?
the comment is a reminder that should replace the usage of `process.env.ENV_KEY` with a more centralized solution which will also allow mocking the environment variable in tests without having to reset the modules.
Currently by using the following part we cannot mock the environment variable (since it's referenced by value in the imported file) unless we reset all modules and import the testing modules in each test.
```typescript
// file: clerkClient.ts
export const SIGN_IN_URL = 'something'

// file: authMiddleware.ts

import { SIGN_IN_URL } from "./clerkClient"
```